Backend/mentii/class_ctrl.py

```diff
@@ -36,6 +21,37 @@ def getActiveClassList(dynamoDBInstance, email=None):
     res = dbUtils.getItem(request, table)
     if res is not None and 'Item' in res:
       classes.append(res['Item'])
+  response.addToPayload('classes', classes)
+  return response
 
+def getClassCodesFromUser(dynamoDBInstance, email=None):
+  classCodes = []
+  if email is None:
+    email = g.authenticatedUser['email']
+  usersTable = dbUtils.getTable('users', dynamoDBInstance)
+  if usersTable is None:
+    MentiiLogging.getLogger().error("Unable to get users table in getClassCodesFromUser")
+    return None
+  #An active class list is the list of class codes that
+  # a user has in the user table.
+  request = {"Key" : {"email": email}, "AttributesToGet": ["classCodes"]}
+  res = dbUtils.getItem(request, usersTable)
+  #Get the class codes for the user.
+  if res is not None and 'Item' in res:
+    classCodes = res['Item']['classCodes']
+  return classCodes
+
+def getPublicClassList(dynamodb, email=None):
+  response = ControllerResponse()
+  classCodes = getClassCodesFromUser(dynamodb, email)
+  classes = []
+  classesTable = dbUtils.getTable('classes', dynamodb)
+  if classesTable is None:
+    MentiiLogging.getLogger().error("Unable to get classes table in getPublicClassList")
+    return None
+  res = classesTable.scan()
+  for pclass in res['Items']:
```

Comment:
There's not error checking on if you were able to get items from the classes table.